Fix simplified page being slow due to very large text

frozenhelium opened the pull request:
- Limit the amount of characters displayed in simplified text
- Add 'Show more' button at the end of simplified text
- Fix unexpected jumps when adding an entry form simplified text
- Add tooltip for ellipsized text in SubNavbar
- Improve styling of edit excerpt popup
- Check commonAncestorContainer before accessing bounding client rect
- Sort entries from server on ascending order of date created
- Newly added entries are added to the bottom
- Use ref instead of state for storing staleIdentifiers and deleteIdentifiers

frozenhelium committed:
Fix simplified page being slow due to very large text

- Limit the amount of characters displayed in simplified text
- Add 'Show more' button at the end of simplified text
- Fix unexpected jumps when adding an entry form simplified text
- Add tooltip for ellipsized text in SubNavbar
- Improve styling of edit excerpt popup

frozenhelium committed:
Use virtualized list in entry list (edit entries page)

tnagorra commented on app/views/Project/EntryEdit/LeftPane/SimplifiedTextView/index.tsx:
```diff
@@ -54,9 +59,32 @@ function SimplifiedTextView(props: Props) {
+    const splits = React.useMemo(() => {
+        // Note: we need to preserve scroll due to
+        // unexpected jumps on entry addition / removal
+        const scrollTop = containerRef.current?.scrollTop;
+        if (isDefined(scrollTop)) {
+            window.setTimeout(() => {
+                if (containerRef.current) {
+                    containerRef.current.scrollTop = scrollTop;
+                }
+            }, 0);
+        }
```
We shouild put this in useEffect

tnagorra commented on app/views/Project/EntryEdit/LeftPane/SimplifiedTextView/styles.css:
```diff
@@ -1,16 +1,22 @@
     white-space: pre-wrap;
     word-break: break-word;
 
+    * {
+        white-space: pre-wrap;
+        word-break: break-word;
+    }
```
Do we need to change this?

frozenhelium replied:
Removed

tnagorra committed:
Preserve scrollTopRef on split change and set on layout effect

- Fix virtualized list styling on "All entries" tab
- Check commonAncestorContainer before accessing bounding client rect
- Use substring instead of substr to get sub-string
- Remove scrollIntoView prop from EntryItem
- Decrease size of EntryItem to 200
- Sort entries from server on ascending order of date created
- Newly added entries are added to the bottom

tnagorra committed:
Use ref instead of state for storing staleIdentifiers and deleteIdentifiers

samshara commented:
DeepScan issues need to be fixed @frozenhelium
